docs/build/getting-started/testing.md

```diff
@@ -0,0 +1,108 @@
+
+# Introduction to Testing in Cadence
+
+Testing is an essential part of smart contract development to ensure the correctness and reliability of your code. Cadence Testing Framework provides a convenient way to write tests for your programs, allowing you to verify the functionality and correctness of your smart contracts.
+
+## Install Flow CLI
+
+The [Flow CLI](../../tools/flow-cli/index.md) is the primary tool for developing, testing, and deploying smart contracts to the Flow network.
+
+If you haven't installed the Flow CLI yet and have [homebrew](https://brew.sh/) installed, simply run `brew install flow-cli`. Alternatively, refer to the Flow CLI [installation instructions](../../tools/flow-cli/install.md).
+
+## Create a new project
+
+In your preferred code editor, create a new directory for your project and navigate to it in the terminal. Then 
+initialize a new Flow project by running the command `flow init`. This will create a `flow.json` file that contains the [project configuration](../../tools/flow-cli/flow.json/configuration.md).
+
+```bash
+mkdir test-cadence
+cd test-cadence
+flow init
+```
+
+## Write a simple smart contract
+
+In your code editor, create a new file called `calculator.cdc` and add the following code:
+
+```cadence calculator.cdc
+access(all) contract Calculator {
+    access(all) fun add(a: Int, b: Int): Int {
+        return a + b;
+    }
+
+    access(all) fun subtract(a: Int, b: Int): Int {
+        return a - b;
+    }
+}
+```
+
+## Add the smart contract to the config
+
+Run `flow config add contract` and add the created `calculator.cdc` contract to the config with the name as `Calculator`. This command adds your contract name and location under the `contracts` key in `flow.json`.
+
+```
+Enter name: Calculate
+Enter contract file location: ./calculator.cdc
+```
+
+## Write test cases
+
+In the same directory, create a new file called `calculator_test.cdc` and add the following code:
+
+```cadence calculator_test.cdc
+import Test
+import "Calculator" // contract name from the previous step
+
+access(all) let calculator = Calculator()
+
+access(all) fun testAdd() {
+  Test.assertEqual(calculator.add(a: 2, b: 3), 5)
```

Comment:
It is a good practice to put the expected value as the left argument of `Test.assertEqual`, e.g.
```cadence
Test.assertEqual(5, calculator.add(a: 3, b: 3))
```
The above would fail with:
```shell
error: assertion failed: not equal: expected: 5, actual: 6
```